code/modules/lighting/lighting_profiler.dm

```diff
@@ -0,0 +1,41 @@
+// Writes lighting updates to the database.
+// FOR DEBUGGING ONLY!
+
+var/DBQuery/lprof_q
+
+/proc/lprof_init()
+	lprof_q = dbcon.NewQuery({"INSERT INTO ss13dbg_lighting (time,type,name,loc_name,x,y,z) 
+		VALUES (:time,:type,:name,:loc_name,:x,:y,:z);"})
+
+/proc/lprof_write(var/atom/movable/obj, var/type = "UNKNOWN")
+	if (!lighting_profiling || !obj || !dbcon.IsConnected())
```

Comment:
Last check should be `!establish_db_connection(dbcon)`. The first thing establish_db_connection does is check IsConnected().